apps/roam/src/components/CreateNodeDialog.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useEffect, useRef, useState } from "react";
+import {
+  Dialog,
+  Classes,
+  InputGroup,
+  HTMLSelect,
+  Button,
+} from "@blueprintjs/core";
+import renderOverlay from "roamjs-components/util/renderOverlay";
+import createDiscourseNode from "~/utils/createDiscourseNode";
+import { OnloadArgs } from "roamjs-components/types";
+import updateBlock from "roamjs-components/writes/updateBlock";
+import { render as renderToast } from "roamjs-components/components/Toast";
+import getUids from "roamjs-components/dom/getUids";
+import { DiscourseNode } from "~/utils/getDiscourseNodes";
+
+export type CreateNodeDialogProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  nodeTypes: DiscourseNode[];
+  defaultNodeType: DiscourseNode;
+  extensionAPI: OnloadArgs["extensionAPI"];
+  blockUid?: string;
+  initialTitle: string;
+};
+
+const CreateNodeDialog = ({
+  isOpen,
+  onClose,
+  nodeTypes,
+  defaultNodeType,
+  extensionAPI,
+  blockUid,
```

Comment:
nit: `blockUid` is rather ambigious.  Maybe something like `sourceBlockUid` or `originBlockUid`